security-command-center/snippets/src/test/java/vtwo/SourceIT.java

```diff
@@ -0,0 +1,116 @@
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import vtwo.source.GetSource;
+import vtwo.source.ListSources;
+import vtwo.source.UpdateFindingSource;
+import vtwo.source.UpdateSource;
+
+@RunWith(JUnit4.class)
+public class SourceIT {
+
+  // TODO: Replace the below variables.
+  private static final String ORGANIZATION_ID = System.getenv("SCC_PROJECT_ORG_ID");
+  private static final String LOCATION = "global";
+  private static Source SOURCE;
+  private static Finding FINDING;
+  private static final int MAX_ATTEMPT_COUNT = 3;
+  private static final int INITIAL_BACKOFF_MILLIS = 120000; // 2 minutes
+  private static ByteArrayOutputStream stdOut;
+
+  @Rule
+  public final MultipleAttemptsRule multipleAttemptsRule = new MultipleAttemptsRule(
+      MAX_ATTEMPT_COUNT,
+      INITIAL_BACKOFF_MILLIS);
+
+  // Check if the required environment variables are set.
+  public static void requireEnvVar(String envVarName) {
+    assertWithMessage(String.format("Missing environment variable '%s' ", envVarName))
+        .that(System.getenv(envVarName))
+        .isNotEmpty();
+  }
+
+  @BeforeClass
+  public static void setUp() throws IOException, InterruptedException {
+    final PrintStream out = System.out;
+    stdOut = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(stdOut));
+
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("SCC_PROJECT_ORG_ID");
+
+    // Create source.
+    SOURCE = Util.createSource(ORGANIZATION_ID);
+
+    String uuid = UUID.randomUUID().toString().split("-")[0];
+    FINDING = Util.createFinding(SOURCE.getName(), "testfindingv2" + uuid, LOCATION,
+        Optional.of("MEDIUM_RISK_ONE"));
+
+    stdOut = null;
+    System.setOut(out);
+  }
+
+  @Test
+  public void testListAllSources() {
+    List<Source> response = ListSources.listSources(ORGANIZATION_ID);
+
+    assertThat(response.stream().map(Source::getName)).contains(SOURCE.getName());
+  }
+
+  @Test
+  public void testGetSource() {
+    Source source = GetSource.getSource(ORGANIZATION_ID, SOURCE.getName().split("/")[3]);
+
+    assertThat(source.getName()).isEqualTo(SOURCE.getName());
+  }
+
+  @Test
+  public void testUpdateSource() {
+    Source source = UpdateSource.updateSource(ORGANIZATION_ID, SOURCE.getName().split("/")[3]);
+
+    assertThat(source.getDisplayName()).contains("Updated Display Name");
+  }
+
+  @Test
+  public void testUpdateFindingSource() {
+    Finding findingUpdated = UpdateFindingSource.updateFinding(ORGANIZATION_ID, LOCATION,
+        SOURCE.getName().split("/")[3], FINDING.getName().split("/")[7]);
+
+    assertThat(findingUpdated).isNotNull();
+    assertThat(findingUpdated).isNotEqualTo(Finding.getDefaultInstance());
```

Comment:
What does `Finding.getDefaultInstance()` imply here?

Reply:
I have updated to a new version, please review